src/Reflection/ReflectionClass.php

```diff
@@ -1238,4 +1244,17 @@ public function removeProperty($propertyName)
 
         return false;
     }
+
+    private function findTypeFromAst($namespace, $locatedSource, $type)
```

Comment:
Wouldn't it make more sense to create the context in `FindTypeForAst` helper itself, as it currently does (also means no BC break in that aspect)?